### DISCUSSION TITLE
Customs Decorator

### DISCUSSION THREAD

**Discussion by zantler** [TypedProject/ts-express-decorators#1606] [Q&A] [answered]
Hi Guys

Im creating a custom decorator for my APIS and i need to use it on my OverrideMiddleware for LogIncomingRequestMiddleware

this is the code for my simple decorator:

```
import { LogIncomingRequestMiddleware } from "@tsed/common";
import { applyDecorators, StoreSet } from "@tsed/core";

export function Ignore(ignore: boolean): Function {
  return applyDecorators(StoreSet(LogIncomingRequestMiddleware, ignore));
}
```

Now, following the documentation int he middleware when im trying to get it from the context the parameter doent exist 

I'll be greatful with any help you can give me

**Comment by zantler** [comment, marked as answer]
Im using version 5.x.x

**Reply by Romakita** [reply]
Hum… where did you use Ignore decorator exactly?

**Reply by Romakita** [reply]
And the middleware must be added at the same time with Use or UseBefore or UseAfter. If you expect to have endpoint info available on LogIncomingRequestMiddlare while it was registered as global middleware, it won’t works. The example show the Store usage with a middleware added on the decorated endpoint ;)
You have missunterstood the example and his goal ^^

See you
Romain

**Reply by zantler** [reply]
So if I understand.. in global middelwares you cant acces to custom decorators, only on endpoint middlewares?

**Reply by Romakita** [reply]
Yes exactly

**Reply by zantler** [reply]
Thanks a lot! However i dont know how but i make it work xD
